frontend/components/analytics/GoogleAnalytics.tsx

```diff
@@ -39,6 +40,7 @@ function GA4() {
    const { query } = router;
 
    const debugMode = GA_DEBUG || query.ga4_debug === 'true';
+   setupMinimumGA4(GTAG_ID, debugMode);
```

Comment:
did you run into issues when it wasn't used?

Reply:
Yeah the events just wouldn't send to GA4 🤷

Reply:
hmm i hadn't run into that. the code in 
```
window.dataLayer = window.dataLayer || [];
               function gtag(){dataLayer.push(arguments);}
               gtag('js', new Date());
               
```
               
should have set it up in time

Reply:
Yep. I played around with that code for a while and couldn't get anything to send. Calling `setupMinimumGA4()` finally made it work with no drawbacks, so I stuck with it.

Reply:
hmmm, im slightly worried about redundancy/spaghetti but we seem to be transitioning to App routing pages which is what `setupMinimumGA4` was made to handle anyway, so its probably to sweep this under the rug

Reply:
@jarstelfox Do you have any insight here? I think we should understand why this works before we run with it.

Reply:
No insights, but I'd try not to call `setupMinimumGA4` it effectively is repeating the code over again

Reply:
we could delete the script in `GoogleAnalytics` to reduce the redundancy

>I think we should understand why this works before we run with it.

I would say the question is more precisely "Why didn't the code that was working before now not work without calling `setupMinimumGA4()`?".

Reply:
Refactored and removed the duplicate js from the `<head>` in [a116c6e](https://github.com/iFixit/react-commerce/pull/2029/commits/a116c6ebcf6e2741ab623a36b4b888c193d8935e).